homeassistant/components/vconnex/config_flow.py

```diff
@@ -0,0 +1,142 @@
+"""Config flow for Vconnex integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from vconnex.api import VconnexAPI
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+
+from .const import (
+    CONF_CLIENT_ID,
+    CONF_CLIENT_SECRET,
+    CONF_PROJECT_NAME,
+    CONF_USER_ID,
+    DEFAULT_ENDPOINT,
+    DOMAIN,
+    DOMAIN_NAME,
+    PROJECT_CODE,
+)
+
+LOGGER = logging.getLogger(__name__)
+
+TOKEN_USER_ID = "userId"
+TOKEN_PROJECT_NAME = "projectName"
+
+
+def validate_input(
+    hass: HomeAssistant, user_input: dict[str, Any]
+) -> dict[str, Any] | None:
```

Comment:
```suggestion
) -> dict[str, Any]:
```
return the info or raise, no need to return `None`